src/ConnectedMode/Persistence/LegacySolutionBindingPathProvider.cs

```diff
@@ -20,36 +20,36 @@
 
 using System;
 using System.IO;
-using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core;
 
 namespace SonarLint.VisualStudio.ConnectedMode.Persistence
 {
     internal class LegacySolutionBindingPathProvider : ISolutionBindingPathProvider
     {
-        private readonly IVsSolution vsSolution;
+        private readonly ISolutionInfoProvider solutionInfoProvider;
         public const string LegacyBindingConfigurationFileName = "SolutionBinding.sqconfig";
 
-        public LegacySolutionBindingPathProvider(IServiceProvider serviceProvider)
+        public LegacySolutionBindingPathProvider(ISolutionInfoProvider solutionInfoProvider)
         {
-            if (serviceProvider == null)
+            if (solutionInfoProvider == null)
             {
-                throw new ArgumentNullException(nameof(serviceProvider));
+                throw new ArgumentNullException(nameof(solutionInfoProvider));
             }
-
-            vsSolution = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
+            this.solutionInfoProvider = solutionInfoProvider;
         }
 
         public string Get()
         {
-            vsSolution.GetSolutionInfo(out var solutionDirectory, out _, out _);
```

Comment:
In this case the old code used a different API call on `IVsSolution` to get the solution directory rather than the solution file path itself.
In the new code I'm getting the full solution path then calling `Path.GetDirectoryName` on it.

I considered adding an extra method to `ISolutionInfoProvider` to return the directory, but haven't done so yet. If there are enough other places in the code that need the solution directory I might add it/an extension method in a separate PR.